nixos/tests/i3wm: fix terminal title (test is broken on master)

hedning opened the pull request:
###### Motivation for this change

#51678 changed terminal titles, breaking the i3wm test. For some reason the gnome3 test is okay even though it looks for the same window title.

cc @yegortimoshenko who did the bash title change

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

hedning committed:
nixos/tests/i3wm: fix terminal title

Probably due to #51678 which makes bash set the terminal title.

hedning commented:
I can update the gnome3 test if it's wanted. I'm guessing it works due to gnome-terminal starting up with  `Terminal` as a title and takes long enough to change for the `waitForWindow` to pick it up.

yegortimoshenko commented:
https://github.com/NixOS/nixpkgs/blob/373be8207ac409f752dbea66d4ff00a653d5b620/nixos/tests/gnome3.nix#L36

Agree that GNOME test should be updated:

```perl
$machine->sleep(2);
$machine->waitForWindow(qr/alice.*machine/);
```

hedning committed:
nixos/tests/gnome3: fix terminal title

The tests passes, but that's just because a race condition where the window is
titled `Terminal` long enough.

hedning commented:
Well it's a bit weird that changing the bash title would break desktop manager tests :laughing: Would be nice getting it decoupled in the long run. Perhaps getting bash to use a predictable title in the tests would be good?

Took a look at the other waitForWindow tests and [`xfce`](https://hydra.nixos.org/build/85712216/download/1/screen.png) and `plasma` should be fine, they both contain the correct strings. The `gnome3-gdm` test will be fixed by #44497 so I'd prefer not making a merge conflict for no reason.
